packages/flutter/lib/src/services/browser_context_menu.dart

```diff
@@ -0,0 +1,72 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+import 'system_channels.dart';
+
+/// Controls the browser's context menu on the web platform.
+///
+/// The context menu is the menu that appears on right clicking or selecting
+/// text in the browser, for example.
+///
+/// On web, by default, the browser's context menu is enabled by default, and
+/// Flutter's context menus are hidden.
+///
+/// On all non-web platforms, this does nothing.
+class BrowserContextMenu {
+  BrowserContextMenu._();
+
+  static final BrowserContextMenu _instance = BrowserContextMenu._();
+
+  /// Whether showing the browser's context menu is enabled.
+  ///
+  /// When true, any event that the browser typically uses to trigger its
+  /// context menu (e.g. right click) will do so. When false, the browser's
+  /// context menu will not show.
+  ///
+  /// It's possible for this to be true but for the browser's context menu to
+  /// not show due to direct manipulation of the DOM. This bool only indicates
+  /// the results of calling [disableContextMenu] and [enableContextMenu] here.
+  ///
+  /// Defaults to true.
+  static bool get enabled => _instance._enabled;
+
+  bool _enabled = true;
+
+  final MethodChannel _channel = SystemChannels.contextMenu;
+
+  /// Disable the browser's context menu.
+  ///
+  /// By default, when the app starts, the browser's context menu is already
+  /// enabled.
+  ///
+  /// See also:
+  ///  * [enableContextMenu], which performs the opposite operation.
+  static Future<void> disableContextMenu() {
```

Comment:
Why does it return a `Future<void>`? Is the developer supposed to be able to do something after the future is resolved?

Reply:
It looks like it's possible that the `enableContextMenu()` is immediately called after this method and when the future returned by this method is resolved `enable` is true but will soon become false, so I'm not sure how useful it would be to have this return a Future.

Reply:
Discussed this offline a bit, but I'm open to tightening this API down if it would remove some tricky corner cases.  Like maybe preventing enable/disable from being called while a Future is still pending, etc.

Reply:
For this example:
```dart
  onSecondaryTapDown: (TapDownDetails details) {
    await BrowserContextMenu.disableContextMenu();
    await Future.delayed(const Duration(seconds: 0));
    BrowserContextMenu.enableContextMenu();
  }
```

is there a way to turn the implementation into one that's hermetic? The code could be written for a library so it doesn't necessarily know what the application is doing with the BrowserContextMenu api and it should turn the flag back to the previous state once it's done,  and you can't really do this:
```dart
final wasEnabled = enabled;
if (wasEnabled) { 
 await BrowserContextMenu.disableContextMenu();
 doStuff();
  BrowserContextMenu.enableContextMenu();
} else { 
  doStuff();
}
```

Reply:
Discussed offline with @LongCatIsLooong.  I'm likely to merge this as-is on Monday unless I can think of any improvements to make.  I plan to watch for issues from users to see how they're using this new API.